File: courseplan.py
class CoursePlan:
    def __init__(self):
        # TODO

    def add_course(self,course):
        # TODO

    def add_requisite(self,course1,course2):
        # TODO

    def find(self):
        # TODO

if __name__ == "__main__":
    c = CoursePlan()
    c.add_course("Ohpe")
    c.add_course("Ohja")
    c.add_course("Tira")
    c.add_course("Jym")
    c.add_requisite("Ohpe","Ohja")
    c.add_requisite("Ohja","Tira")
    c.add_requisite("Jym","Tira")
    print(c.find()) # [Ohpe,Jym,Ohja,Tira]
    c.add_requisite("Tira","Tira")
    print(c.find()) # None

File: Java_template/CoursePlan.java
import java.util.*;

public class CoursePlan {
    public void addCourse(String course) {
        // TODO
    }

    public void addRequisite(String course1, String course2) {
        // TODO
    }

    public ArrayList<String> find() {
        // TODO
    }

    public static void main(String[] args) {
        CoursePlan c = new CoursePlan();
        c.addCourse("Ohpe");
        c.addCourse("Ohja");
        c.addCourse("Tira");
        c.addCourse("Jym");
        c.addRequisite("Ohpe","Ohja");
        c.addRequisite("Ohja","Tira");
        c.addRequisite("Jym","Tira");
        System.out.println(c.find()); // [Ohpe,Jym,Ohja,Tira]
        c.addRequisite("Tira","Tira");
        System.out.println(c.find()); // null
    }
}
